syntax = "proto3";

option go_package          = "github.com/cosmos/cosmos-sdk/streaming/plugins/abci/proto";
option java_package        = "network.cosmos.sdk.streaming.grpc";
option java_multiple_files = true;

// PutRequest is used for storing ABCI request and response
// and Store KV data for streaming to external grpc service.
message PutRequest {
  int64 block_height  = 1;
  bytes req           = 2;
  bytes res           = 3;
  bytes store_kv_pair = 4;
}

message Empty {}

service ABCIListenerService {
  rpc ListenBeginBlock(PutRequest) returns (Empty);
  rpc ListenEndBlock(PutRequest) returns (Empty);
  rpc ListenDeliverTx(PutRequest) returns (Empty);
  rpc ListenStoreKVPair(PutRequest) returns (Empty);
}
